pkg/taskrunmetrics/metrics.go

```diff
@@ -0,0 +1,160 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package taskrunmetrics
+
+import (
+	"context"
+	"sync"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/metrics"
+)
+
+const anonymous = "anonymous"
+
+var (
+	sgCountView *view.View
+
+	sgCount = stats.Float64("taskrun_signed_total",
+		"Total number of signed taskruns",
+		stats.UnitDimensionless)
+
+	plCount = stats.Float64("taskrun_payload_uploaded_total",
+		"Total number of uploaded payloads for taskruns",
+		stats.UnitDimensionless)
+
+	plCountView *view.View
+
+	stCount = stats.Float64("taskrun_payload_stored_total",
+		"Total number of stored payloads for taskruns",
+		stats.UnitDimensionless)
+
+	stCountView *view.View
+
+	mrCount = stats.Float64("taskrun_marked_signed_total",
+		"Total number of objects marked as signed for taskruns",
+		stats.UnitDimensionless)
+
+	mrCountView *view.View
+)
+
+// Recorder is used to actually record TaskRun metrics
+type Recorder struct {
+	mutex       sync.Mutex
+	initialized bool
+}
+
+// We cannot register the view multiple times, so NewRecorder lazily
+// initializes this singleton and returns the same recorder across any
+// subsequent invocations.
+var (
+	once           sync.Once
+	r              *Recorder
+	errRegistering error
+)
+
+// NewRecorder creates a new metrics recorder instance
+// to log the TaskRun related metrics
+func NewRecorder(ctx context.Context) (*Recorder, error) {
+	once.Do(func() {
+		r = &Recorder{
+			initialized: true,
+		}
+
+		errRegistering = viewRegister()
+
+		if errRegistering != nil {
+			r.initialized = false
+			return
+		}
+	})
+
+	return r, errRegistering
+}
+
+func viewRegister() error {
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
+
+	sgCountView = &view.View{
+		Description: sgCount.Description(),
+		Measure:     sgCount,
+		Aggregation: view.Count(),
+	}
+
+	plCountView = &view.View{
+		Description: plCount.Description(),
+		Measure:     plCount,
+		Aggregation: view.Count(),
+	}
+
+	stCountView = &view.View{
+		Description: stCount.Description(),
+		Measure:     stCount,
+		Aggregation: view.Count(),
+	}
+
+	mrCountView = &view.View{
+		Description: mrCount.Description(),
+		Measure:     mrCount,
+		Aggregation: view.Count(),
+	}
+	return view.Register(
+		sgCountView,
+		plCountView,
+		stCountView,
+		mrCountView,
+	)
+}
+
+func viewUnregister() {
+	view.Unregister(
+		sgCountView, plCountView, stCountView, mrCountView,
+	)
+}
+
+func (r *Recorder) RecordCountMetrics(MetricType string, ctx context.Context) {
+	logger := logging.FromContext(ctx)
+	logger.Infof("Recording payload count metrics for taskrun ", ctx)
+
+	if !r.initialized {
+		logger.Errorf("ignoring the metrics recording as recorder not initialized ")
+	}
+	switch mt := MetricType; mt {
+	case "sgcount":
```

Comment:
We can define these constants in pkg/chains and then reuse them here instead of using string.

Reply:
Created constants.go and used these constants